### PR TITLE
[formrecognizer] preview 1 design changes

### LINKED ISSUE #10189: [FormRecognizer] Implement sync APIs
- [ ] Receipt
- [ ] Layout
- [ ] Custom
   - [ ] Training labeled/unlabeled
   - [ ] Analyze/extract form
   - [ ] List models
   - [ ] Get models summary
   - [ ] Delete model
   - [ ] Get custom model (2 methods - labeled and unlabeled)

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves #10189 #10190 #10244 

Running/working on tests in this PR: #10711

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 1]
```diff
@@ -40,134 +42,219 @@ class FormRecognizerClient(object):
...
+        def analyze_callback(raw_response, _, headers):  # pylint: disable=unused-argument
+            analyze_result = self._client._deserialize(AnalyzeOperationResult, raw_response)
+            return prepare_form_result(analyze_result, model_id)
+
+        deserialization_callback = cls if cls else analyze_callback
+        return self._client.begin_analyze_with_custom_model(
+            file_stream=stream,
+            model_id=model_id,
+            include_text_details=include_text_content,
+            content_type=content_type,
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:206]: just want to make sure that you can pass an enum all the way through

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -38,6 +38,7 @@
 /sdk/servicebus/ @annatisch @yunhaoling @YijunXieMS @KieranBrantnerMagee
 /sdk/synapse/ @aim-for-better @idear1203
 /sdk/textanalytics/ @kristapratico @iscai-msft
+/sdk/formrecognizer/ @kristapratico
```
[the remark this answers is not in the document]
Comment (kristapratico) [reply .github/CODEOWNERS:41]: Will do in my tests PR :)

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 3]
```diff
@@ -40,134 +42,219 @@ class FormRecognizerClient(object):
...
         :return: LROPoller
-        :rtype: ~azure.core.polling.LROPoller[list[~azure.ai.formrecognizer.ExtractedReceipt]]
+        :rtype: ~azure.core.polling.LROPoller[list[~azure.ai.formrecognizer.USReceipt]]
         :raises: ~azure.core.exceptions.HttpResponseError
         """
-        if not isinstance(url, six.string_types):
-            raise TypeError("Call begin_extract_receipts() to analyze a receipt from a stream.")
 
-        include_text_details = kwargs.pop("include_text_details", False)
+        include_text_content = kwargs.pop("include_text_content", False)
```
[the remark this answers is not in the document]
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:109]: We agreed during our client side validation meeting that we would let the service throw the exception in this case

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_response_handlers.py [hunk 4]
```diff
@@ -138,7 +138,7 @@ def prepare_unlabeled_result(response):
             ),
             fields=unlabeled_fields,
             form_type="form-" + str(page.cluster_id) if page.cluster_id is not None else None,
-            pages=form_pages[page.page-1]
+            pages=[form_pages[page.page-1]]
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_response_handlers.py:141]: My sample works with this change, thanks

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 5]
```diff
@@ -514,65 +587,84 @@ def _from_generated(cls, model):
...
+            training_documents=TrainingDocumentInfo._from_generated(model.train_result)
+            if model.train_result else None
         )
 
 
-class TrainingInfo(object):
-    """Training information about the custom model.
+class CustomFormSubModel(object):
+    """Represents the submodel with model accuracy,
+    any errors while training the model, fields that were
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:600]: should we delete the errors section from this description?

Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:600]: Ah yes, I haven't updated the docstrings after all the design changes. Thanks for point this out!

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 6]
```diff
@@ -609,178 +701,16 @@ def __init__(self, **kwargs):
 
     @classmethod
     def _from_generated(cls, err):
-        return [cls(code=error.code, message=error.message) for error in err] if err else None
+        return [cls(code=error.code, message=error.message) for error in err] if err else []
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:704]: out of curiosity, why the difference in returning `[]` instead of `None`?

Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:704]: I tried to match what the service does here. they seem to send a list of errors, even if there are none (so empty list)

Comment (maririos) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:704]: interesting. Thank you!

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py (hunk 5 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py]: updated

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 7]
```diff
@@ -94,16 +101,16 @@ def begin_recognize_receipts(self, stream, **kwargs):
...
 
         :param url: The url of the receipt. Currently only supports US sales receipts.
         :type url: str
         :keyword bool include_text_content: Include text lines and text content references in the result.
-        :return: LROPoller
+        :return: An instance of an LROPoller. Call `result()` on the poller
+            object to return the result of the long running operation.
         :rtype: ~azure.core.polling.LROPoller[list[~azure.ai.formrecognizer.USReceipt]]
-        :raises: ~azure.core.exceptions.HttpResponseError
+        :raises ~azure.core.exceptions.HttpResponseError:
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:113]: Why did you move the colon to the end? If you're adding it to the end why doesn't rtype have it at the end

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 8]
```diff
@@ -172,19 +183,24 @@ def begin_recognize_content_from_url(self, url, **kwargs):
 
     @distributed_trace
     def begin_recognize_custom_forms(self, model_id, stream, **kwargs):
-        # type: (str, IO[bytes], Any) -> LROPoller
-        """Analyze Form.
+        # type: (str, Union[bytes, IO[bytes]], Any) -> LROPoller
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:186]: I'd add somewhere in the docstring that the form should be of the same type as the forms the custom model was trained with

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 9]
```diff
@@ -172,19 +183,24 @@ def begin_recognize_content_from_url(self, url, **kwargs):
...
+        """Analyze a custom form with a model trained with or without labels.
+        The input document must be of one of the supported content types - 'application/pdf',
+        'image/jpeg', 'image/png' or 'image/tiff'.
 
         :param str model_id: Model identifier.
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:191]: I'd call this "Custom model identifier" to keep it consistent with how we're addressing it elsewhere

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 10]
```diff
@@ -172,19 +183,24 @@ def begin_recognize_content_from_url(self, url, **kwargs):
...
         :param stream: .pdf, .jpg, .png or .tiff type file stream.
         :type stream: stream
         :keyword bool include_text_content: Include text lines and element references in the result.
-        :keyword str content_type: Media type of the body sent to the API. For options,
+        :keyword str content_type: Media type of the body sent to the API. Content-type is
+            auto-detected, but can be overridden by passing this keyword argument. For options,
             see :class:`~azure.ai.formrecognizer.FormContentType`.
-        :return: LROPoller
+        :return: An instance of an LROPoller. Call `result()` on the poller
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:198]: I'd make this more specific to "Call `result()` ... to return the list of `RecognizedForm`s

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py [hunk 11]
```diff
@@ -54,12 +57,12 @@ def __init__(self, endpoint, credential, **kwargs):
...
     def begin_training(self, training_files, use_labels=False, **kwargs):
         # type: (str, Optional[bool], Any) -> LROPoller
-        """Create and train a custom model. The request must include a source parameter that is an
+        """Create and train a custom model. The request must include a training_files parameter that is an
         externally accessible Azure storage blob container Uri (preferably a Shared Access Signature Uri).
         Models are trained using documents that are of the following content type - 'application/pdf',
         'image/jpeg', 'image/png', 'image/tiff'. Other type of content in the container is ignored.
 
-        :param str training_files: An Azure Storage blob container URI.
+        :param str training_files: An Azure Storage blob container SAS URI.
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:65]: nit: container's

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 12]
```diff
@@ -553,29 +551,31 @@ def _from_generated(cls, cell, page, read_result):
 
 
 class CustomFormModel(object):
-    """Represents a custom model returned from training.
+    """Represents a model trained from custom forms.
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:554]: trained on custom forms?

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 13]
```diff
@@ -553,29 +551,31 @@ def _from_generated(cls, cell, page, read_result):
...
+        Date and time (UTC) when model training completed.
     :ivar list[~azure.ai.formrecognizer.CustomFormSubModel] models:
-        A list of submodels which contains the fields trained on.
+        A list of submodels, each of which extract fields from
+        a different type of form.
     :ivar list[~azure.ai.formrecognizer.FormRecognizerError] errors:
         List of any training errors.
     :ivar ~azure.ai.formrecognizer.TrainingDocumentInfo training_documents:
-        Training document information about the custom model.
+         Meta-data about each of the documents used to train the model.
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:571]: nit: metadata

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py (hunk 7 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:113]: necessary for the reference to resolve. we do this in text analytic as well

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py (hunk 12 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:554]: "On" doesn't sound right to me. Anne suggested this change for the docstring so i'm going to keep it.
